src/query/service/src/sessions/session_privilege_mgr.rs

```diff
@@ -267,14 +268,16 @@ impl SessionPrivilegeManager for SessionPrivilegeManagerImpl {
 
     #[async_backtrace::framed]
     async fn has_ownership(&self, object: &OwnershipObject) -> Result<bool> {
-        let role_mgr = RoleCacheManager::instance();
+        let user_mgr = UserApiProvider::instance();
         let tenant = self.session_ctx.get_current_tenant();
-
-        // if the object is not owned by any role, then considered as ACCOUNT_ADMIN, the normal users
-        // can not access it unless ACCOUNT_ADMIN grant relevant privileges to them.
-        let owner_role_name = match role_mgr.find_object_owner(&tenant, object).await? {
-            Some(owner_role) => owner_role.name,
+        let owner_role_name = match user_mgr.get_ownership(&tenant, object).await? {
             None => BUILTIN_ROLE_ACCOUNT_ADMIN.to_string(),
+            Some(owner) => match user_mgr.get_role(&tenant, owner.role).await {
```

Comment:
this match statement could be shorten as `user_mgr.get_role(&tenant, owner.role).map(|r| r.name)?` ?